app/services/IncomeSourceDetailsService.scala

```diff
@@ -224,5 +230,24 @@ class IncomeSourceDetailsService @Inject()(val incomeTaxViewChangeConnector: Inc
       }
     }.toEither
   }
+
+  def getLongAddressFromBusinessAddressDetails(address: Option[AddressModel]): Option[String] = {
+
+    address match {
+      case Some(address) => {
+        val nonNoneFields = List(
+          Some(address.addressLine1),
+          address.addressLine2,
+          address.addressLine3,
+          address.addressLine4,
+          address.postCode,
+          Some(address.countryCode)
+        ).flatten
+
+        Some(nonNoneFields.mkString("<br>"))
```

Comment:
remove any UI related formatting from the service layer, 
you can do these formatting within views/templates

Reply:
so move this method to a code block? If I do this i need to add the whole address to the view model. Can I use AddressModel in the view model or do i need to make a put the address into a separate view model called addressViewModel or can I just add each of the fields to the view model separately?

Reply:
This should be fine for next field Option[AddressModel] "travel" all way up to the view as a part of view model; 
You can create a generic method/re-usable block to render address as it was discussed (as its always easy to change this code there, rather than search for this somewhere in the scala code) + you can also promote this code to be a component if team happy to re-use it in other places (but this is more optional I would say);

Reply:
yeah I guess... I just thought it would be better to add a method to just do it in service as it makes it much easier and there's less to go wrong in service but i will change it